execute/report/builder.go

```diff
@@ -148,15 +156,37 @@ func (b *execReportBuilder) Add(
 	}
 
 	b.execReports = append(b.execReports, execReport)
+	b.commitReports = append(b.commitReports, updatedReport)
 
 	return updatedReport, nil
 }
 
-func (b *execReportBuilder) Build() ([]cciptypes.ExecutePluginReportSingleChain, error) {
+func (b *execReportBuilder) Build() (
+	[]cciptypes.ExecutePluginReportSingleChain, []exectypes.CommitData, error,
+) {
+	if len(b.execReports) != len(b.commitReports) {
+		return nil, nil, fmt.Errorf(
+			"expected the same number of exec and commit reports, got %d and %d",
+			len(b.execReports),
+			len(b.commitReports),
+		)
+	}
+
+	// Check if limiting is required.
+	if b.maxSingleChainReports != 0 && uint64(len(b.execReports)) > b.maxSingleChainReports {
+		b.lggr.Infof(
+			"limiting number of reports to maxReports from %d to %d",
+			len(b.execReports),
+			b.maxSingleChainReports,
+		)
+		b.execReports = b.execReports[:b.maxSingleChainReports]
+		b.commitReports = b.commitReports[:b.maxSingleChainReports]
+	}
```

Comment:
Max single chain report limiting during final build.

I did this here because in the future this would generate multiple OCR reports. I'm imagining that would be done by analyzing a bunch of ExecutePluginReportSingleChain.